Dockerfile

```diff
@@ -1,27 +1,46 @@
-FROM python:3.10-slim AS builder
+# syntax=docker/dockerfile:1
+
+FROM python:3.10-slim@sha256:9956522e7eafd57e3e7bb4b102d56f02882924019867cd2036c1a7c3ee56b174 AS builder
 
 ARG POETRY_VERSION=1.4.2
 RUN pip -V
-RUN pip install poetry==$POETRY_VERSION
+# hadolint ignore=DL3042
+RUN --mount=type=cache,sharing=locked,id=pipcache,mode=0777,target=/root/.cache/pip \
+    pip install --no-compile poetry==$POETRY_VERSION
 
-WORKDIR /src
+WORKDIR /app
 
-COPY pyproject.toml poetry.lock README.md pylintrc ./
 COPY marge/ ./marge/
-RUN poetry export -o requirements.txt && \
-  poetry build
-
-
-FROM python:3.10-slim
-
-RUN apt-get update && apt-get install -y \
-  git-core build-essential \
-  && \
-  rm -rf /var/lib/apt/lists/*
-
-COPY --from=builder /src/requirements.txt /src/dist/marge-*.tar.gz /tmp/
-
-RUN pip install --no-deps -r /tmp/requirements.txt && \
-  pip install /tmp/marge-*.tar.gz
+RUN --mount=type=bind,source=poetry.lock,target=poetry.lock \
+    --mount=type=bind,source=README.md,target=README.md \
+    --mount=type=bind,source=pylintrc,target=pylintrc \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    --mount=type=cache,sharing=locked,id=pipcache,mode=0777,target=/root/.cache/pip \
+    poetry export -o requirements.txt \
+    && pip wheel --no-deps --wheel-dir /app/wheels -r requirements.txt \
+    && poetry build --quiet --no-ansi --no-interaction --format=sdist
+
+FROM python:3.10-slim@sha256:9956522e7eafd57e3e7bb4b102d56f02882924019867cd2036c1a7c3ee56b174 AS runtime
+
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    apt-get update -q \
+    && apt-get install -yq --no-install-recommends \
+    git=1:2.39.2-1.1 \
```

Comment:
I'm not sure how Debian(?) normally handle things like this, will this exact version be available forever and ever?

Reply:
not sure about this one!